### PR TITLE
chore(setup): Containerize repository

### LINKED ISSUE #302: Containerize repository
Add a Dockerfile and a guide to setup Docker for the repository

### PR DESCRIPTION
**Pull request by tkrugg**
| Q                 | A
| ----------------- | ----------
| Bug fix?          | no
| New feature?      | no   
| BC breaks?        | no     
| Related Issue     | Fix  #302
| Need Doc update   | no

## Describe your change

This commit adds:
- Dockerfile
- Specific instructions for setting up and running the tests with Docker
- Adds link to DOCKER_README.md in main README.md

## What problem is this fixing?

#302

### CODE REVIEW

File: DOCKER_README.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+```
+
+However, we advise you to export them. That way, you can use [Docker's shorten syntax](https://docs.docker.com/engine/reference/commandline/run/#set-environment-variables--e---env---env-file) to set your variables.
+
+```bash
+export ALGOLIA_APP_ID=XXXXXX 
+export ALGOLIA_API_KEY=XXX 
+export ALGOLIA_SEARCH_API_KEY=XXXX 
+
+docker run -it --rm --env ALGOLIA_APP_ID --env ALGOLIA_API_KEY -v $PWD:/code -w /code algolia-scout-extended bash
```
Comment (chloelbn) [inline DOCKER_README.md:60]:
```suggestion
docker run -it --rm --env ALGOLIA_APPLICATION_ID --env ALGOLIA_API_KEY -v $PWD:/code -w /code algoliasearch-django bash
```

File: DOCKER_README.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+```bash
+docker run -it --rm --env ALGOLIA_APP_ID=XXXXXX \
+                    --env ALGOLIA_API_KEY=XXXXXXXXXXXXXX \
+                    -v $PWD:/code -w /app algoliasearch-django bash
+```
+
+However, we advise you to export them. That way, you can use [Docker's shorten syntax](https://docs.docker.com/engine/reference/commandline/run/#set-environment-variables--e---env---env-file) to set your variables.
+
+```bash
+export ALGOLIA_APP_ID=XXXXXX
```
Comment (chloelbn) [inline DOCKER_README.md:56]:
```suggestion
export ALGOLIA_APPLICATION_ID=XXXXXX 
```

File: DOCKER_README.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+                    --env ALGOLIA_API_KEY=XXXXXXXXXXXXXX \
+                    -v $PWD:/code -w /app algoliasearch-django bash
+```
+
+However, we advise you to export them. That way, you can use [Docker's shorten syntax](https://docs.docker.com/engine/reference/commandline/run/#set-environment-variables--e---env---env-file) to set your variables.
+
+```bash
+export ALGOLIA_APP_ID=XXXXXX 
+export ALGOLIA_API_KEY=XXX 
+export ALGOLIA_SEARCH_API_KEY=XXXX
```
Comment (chloelbn) [inline DOCKER_README.md:58]: is it necessary?

File: DOCKER_README.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+```bash
+docker build -t algoliasearch-django .
+```
+
+## Run the image
+
+You need to provide a few environment variables at runtime to be able to run the image and the test suite.
+
+```bash
+docker run -it --rm --env ALGOLIA_APP_ID=XXXXXX \
```
Comment (chloelbn) [inline DOCKER_README.md:48]:
```suggestion
docker run -it --rm --env ALGOLIA_APPLICATION_ID=XXXXXX \
```

**Review by chloelbn — CHANGES_REQUESTED** [review]

File: DOCKER_README.md (hunk 3 above)
Comment (tkrugg) [reply DOCKER_README.md:58]:
```suggestion
```

**Comment by chloelbn** [comment]
Most of the tests pass but I still get this error in the second part
```
ERROR: Could not install packages due to an OSError: [Errno 39] Directory not empty: 'management'
```

**Comment by tkrugg** [comment]
> Most of the tests pass but I still get this error in the second part
> 
> ```
> ERROR: Could not install packages due to an OSError: [Errno 39] Directory not empty: 'management'
> ```
Hi  @chloelbn I'm not sure how you got the error. Could you explicit the steps to reproducing the issue?

**Review by chloelbn — APPROVED** [review]

**Pull request merged**